dev/src/bulk-writer.ts

```diff
@@ -271,6 +282,18 @@ class BulkCommitBatch extends WriteBatch {
   }
 }
 
+/**
+ * Used to represent a buffered BulkWriterOperation.
+ *
+ * @private
+ */
+class BufferedOperation {
```

Comment:
Can you just enqueue BulkWriterOperation and make send() public?

Reply:
I tried that approach, but that requires storing `enqueueOnBatchCallback` onto `BulkWriterOperation`, which doesn't make sense conceptually.